File: app/src/main/java/com/example/pam_project/features/lists/list/ListView.java
package com.example.pam_project.features.lists.list;

import com.example.pam_project.features.categories.list.CategoryInformation;

import java.util.List;

public interface ListView {

    void launchFtu();

    void bindCategories(List<CategoryInformation> model);

    void bindLists(final List<ListInformation> model);

    void showAddList();

    void bindSearchedLists(String searchQuery);

    void unbindSearchedLists();

    void showFilterDialog();

    void showSortByDialog();

    void showManageCategories();

    void showListContent(final long id);

    void showEmptyMessage();

    void onListsReceivedError();

    void onCategoriesReceivedError();
}

File: app/src/main/java/com/example/pam_project/features/lists/list/ListView.kt
package com.example.pam_project.features.lists.list

import com.example.pam_project.features.categories.list.CategoryInformation

interface ListView {
    fun launchFtu()
    fun unFocusSearch()
    fun bindCategories(model: List<CategoryInformation?>?)
    fun bindLists(model: List<ListInformation?>?)
    fun showAddList()
    fun bindSearchedLists(searchQuery: String?)
    fun unbindSearchedLists()
    fun showFilterDialog()
    fun showSortByDialog()
    fun showManageCategories()
    fun showListContent(id: Long)
    fun showEmptyMessage()
    fun onListsReceivedError()
    fun onCategoriesReceivedError()
    fun showAboutSection()
}